libraries/AP_RangeFinder/AP_RangeFinder_Backend_Serial.h

```diff
@@ -31,4 +31,9 @@ class AP_RangeFinder_Backend_Serial : public AP_RangeFinder_Backend
 
     // maximum time between readings before we change state to NoData:
     virtual uint16_t read_timeout_ms() const { return 200; }
+
+    // 0 is no return value, 100 is perfect.  false means signal_quality
+    // is not available and quality_pct set to SIGNAL_QUALITY_NODATA.
+    virtual bool get_signal_quality_pct(int8_t &quality_pct) const
+    { quality_pct = RangeFinder::SIGNAL_QUALITY_NODATA; return false; }
```

Comment:
Why keep this for serial? I scanned the code, and AFAICT only 2 serial drivers actually touched this (BLPing and LightWareSerial) so the amount of churn is small. I can't find any callers to get_signal_quality_pct except AP_RangeFinder_Backend_Serial::update(). I suppose if somebody has a fork that we don't know about it could be a problem. A question for the maintainers?

P.S. if we end up keeping this it should be marked WARNED_IF_UNUSED. This is the macro used by AP to keep all compilers happy.

Reply:
I didn't explain this very well in the original comment. RangeFinderBackendSerial implements update(). This update() routine updates the range finder state and uses virtual method get_reading() to get a distance measurement from all the serial drivers that inherit from RangeFinderBackendSerial. With this PR, this update routine also needs to get a signal_quality value to put in the state. The existing virtual function get_signal_quality_pct() is usurped for this purpose. Note that  get_signal_quality_pct() is now a virtual function on RangeFinderBackendSerial and not on RangeFinderBackend. So get_signal_quality_pct()is no longer visible to clients of RangeFinderBackend.

Reply:
Oh, I see.